### PR TITLE
Hook up new pages to profile

### PR DESCRIPTION
**Pull request by cristipaval**
### Details
Hooks up the new Display name, Pronouns and Timezone pages to Profile page

### Fixed Issues
$ https://github.com/Expensify/App/issues/12952

### Tests and QA Steps
1. Go to Settings -> Profile
2. Check that the Profile Page looks like in the video recordings
3. Tap on **Display name** and verify that you are directed to the new page to edit your name.
4. Update the name and go back. Verify that the name is updated on the Profile Page.
5. Do the same with pronouns.
6. The same with the timezone
7. Go back to settings and sign out
8. Sign in back and verify that the name, pronouns and timezone are as they were set at the steps 4-6.

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open the app in 2 platforms (or one in normal chrome, one in incognito), log in to the same account in both
2. Navigate to the `Settings -> Profile` in both devices
3. Go offline in one, update your name, pronouns and timezone settings
4. Verify it looks like it worked correctly on that device
5. Go online, verify your name, pronouns and timezone update on the other device as well

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/18078393/206020157-3ccc788f-388d-4deb-a528-bc4ef63088bf.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/18078393/206020325-3a7392ac-3738-4287-9d68-a2d87aa213bd.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/18078393/206020419-64bd659c-095c-4146-a487-07e26b7035a5.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/18078393/206020475-bf0b3f52-58e3-4885-852f-5078e4742796.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/18078393/206020524-dadc60bd-cc7a-49b9-aea8-33081294e933.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/18078393/206020590-90c277e0-385d-400d-ac15-47cd2054840a.mov

</details>

### CODE REVIEW

File: src/libs/actions/PersonalDetails.js
Comment (cristipaval) [inline src/libs/actions/PersonalDetails.js:151]: I created [this](https://github.com/Expensify/Expensify/issues/247222) GH issue to watch this api command and remove when possible.

Comment (Beamanator) [reply src/libs/actions/PersonalDetails.js:151]: Hmm this cleanup is supposed to take place in this issue: https://github.com/Expensify/Expensify/issues/231919

You can see my whole tracking list of issues for Account Settings here: https://github.com/Expensify/Expensify/issues/182122

File: src/pages/settings/Profile/ProfilePage.js
Comment (Beamanator) [inline src/pages/settings/Profile/ProfilePage.js:156]: I like this, I thinkkkkk there's also `displayName` in `currentUserPersonalDetails`, right? Which do you think is better?

File: src/pages/settings/Profile/ProfilePage.js
Comment (Beamanator) [inline src/pages/settings/Profile/ProfilePage.js:171]: Was this needed for some style fixes or something?

**Review by Beamanator — COMMENTED** [review]
Overall the code is looking great! I haven't tested the PR yet though cuz I usually do that only after the author included screenshots / vids I can compare my tests to :D

One quick comment: Often when I have a PR that includes adding new styles, I think about: Are these new styles actually necessary? Or do other pages do something I can also take advantage of to produce the same effects?

I am just wondering if they're all needed or if something like keeping `ScreenWrapper` would have the safe effect? 

Anyway overall great job dude 👍

File: src/pages/settings/Profile/ProfilePage.js
Comment (cristipaval) [reply src/pages/settings/Profile/ProfilePage.js:156]: Woah, I missed `displayName` in `currentUserPersonalDetails`. It is definitely better.

File: src/pages/settings/Profile/ProfilePage.js
Comment (cristipaval) [reply src/pages/settings/Profile/ProfilePage.js:171]: I wanted to make the content scrollable. I realised that when I inspect the App on web, I can't access all menu items on the ProfilePage if the dev tools are rendered at the bottom of the window. Doesn't it make sense?
Screenshot 2022-12-06 at 14 04 25

File: src/pages/settings/Profile/ProfilePage.js
Comment (cristipaval) [reply src/pages/settings/Profile/ProfilePage.js]: fixed.

File: src/pages/settings/Profile/ProfilePage.js
Comment (Beamanator) [reply src/pages/settings/Profile/ProfilePage.js:171]: Ooh well i definitely think it's good you caught this - about it not being able to scroll before 😅 (in case the user has a very short screen I guess?)

But I thinkkk you should put the `ScrollVew` as a sibling of `HeaderWithCloseButton` like we have here: https://github.com/Expensify/App/blob/fa19a2d5e6f27dbc0f72ead48afddcd85e1d20b5/src/pages/ReportDetailsPage.js#L110-L116

Right?

Comment (cristipaval) [reply src/pages/settings/Profile/ProfilePage.js:171]: you're right, thanks

**Comment by shawnborton** [comment]
I responded via email but I don't think it came through here. 

For this screen here:
image


Can you make the Email address and Phone number headings look identical to Display name, Pronouns, etc? I believe that should be a reusable style (the combination of using fontSizeLabel and color.textSupporting) and we should standardize on that heading style elsewhere too (though we don't need to address that in this PR).

**Comment by cristipaval** [comment]
Thank you @shawnborton ! I added you as a reviewer to check my video recordings and see if you're ok with how the page looks.

**Comment by shawnborton** [comment]
Looks great, thanks Cristi!

**Comment by sobitneupane** [comment]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/25876548/206257178-a17bd0f0-fa02-46fb-82f2-28b647e70dd0.mov


</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/25876548/206257442-4f93be84-e45c-4bcc-985e-6030dce45a0e.mov




</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/25876548/206257812-824d00d6-30c5-4872-b669-c19bd3c24dfa.mov




</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/25876548/206257393-b5aafb56-09c2-4544-b93f-f5538ed1ba7c.mov




</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/25876548/206257527-3c3f70ad-3f9d-4ab6-a356-4fc2b4285207.mov




</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/25876548/206257593-5176b32a-287c-44af-a03b-207663c825e2.mov




</details>

**Comment by sobitneupane** [comment]
>Offline tests
>1. Open the app in 2 platforms (or one in normal chrome, one in incognito), log in to the same account in both
>2. Navigate to the Settings -> Profile in both devices
>3. Go offline in one, update your name, pronouns and timezone settings
>4. Verify it looks like it worked correctly on that device
>5. Go online, verify your name, pronouns and timezone update on the other device as well

Either online or offline, changes made from one platform or web is only reflected after refresh in the web/mWeb. 

https://user-images.githubusercontent.com/25876548/206265105-603c4109-4daa-4675-9f1c-01499ed6bd5b.mov

**Comment by shawnborton** [comment]
> The margin/gap between Phone number label and phone number is large compared to others.

Yeah, I agree with that.

**Comment by cristipaval** [comment]
> > The margin/gap between Phone number label and phone number is large compared to others.
> 
> Yeah, I agree with that.

That's because the **phone number** is wrapped and vertically aligned with the **Resend** button. If we lifted the phone number to the top a bit, I think it would look weird to have no alignment with the Resend button and to have a bigger gap between the phone number and the note below.
@shawnborton What do you think?
I think we could do nothing for now because:
1. This state is seen shortly by the user. The user adds the phone number and it takes up to 1 minute to validate it. The Resend button and the note below disappear afterwards.
2. This is going to be changed soon. The new Contact Method page is on hold for the moment but it is part of N7.

**Comment by sobitneupane** [comment]
@cristipaval Both the browsers(regular and incognito) are online.

**Comment by sobitneupane** [comment]
@cristipaval Both the browsers(regular and incognito) are online.

**Comment by cristipaval** [comment]
> @cristipaval Both the browsers(regular and incognito) are online.

Yes, but in your video you turn your wi-fi off completely on your macbook, and then when you turn it back on, both chrome windows come online at the same time. Please turn Wi-Fi off and then turn it back on only the window where you do the profile changes. Then check that the other window gets the Puhser updates and updates the profile settings accordingly.

**Comment by sobitneupane** [comment]
@cristipaval I have not turned off wifi at all.

**Comment by cristipaval** [comment]
> @cristipaval I have not turned off wifi at all.

Oh right, I see. Is this always reproducible on your end? I can't reproduce on my side.

**Comment by sobitneupane** [comment]
@cristipaval It is always reproducible. I was able to reproduce it again.

**Comment by cristipaval** [comment]
@youssef-lr @Beamanator Could you guys test and see if you can reproduce @sobitneupane's issue?

**Comment by youssef-lr** [comment]
@cristipaval On it.

**Comment by shawnborton** [comment]
> If we lifted the phone number to the top a bit, I think it would look weird to have no alignment with the Resend button and to have a bigger gap between the phone number and the note below.

Good point, I vote we do nothing and leave as-is. Or, we could make the resend button use a smaller button style.

**Comment by cristipaval** [comment]
Good catch @youssef-lr! There is an open [issue](https://github.com/Expensify/Expensify/issues/244501) meant to fix pusher when testing dev builds hitting the staging api.

**Review by sobitneupane — DISMISSED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by Beamanator — CHANGES_REQUESTED** [review]
One thing we might want to address is "Phone number" label appearing over "Add phone number" - Idk if this is a big deal since we're going to replace all that once #passwordless is unblocked and we implement adding / viewing / deleting contact methods, but... In the meantime I think I'd like to see the "Phone number" label hidden when the "Add Phone number" button is shown, what do y'all think?

(Oh also I couldn't reproduce the pusher delays in my VM, and I had trouble signing in to the staging / production envs so I'll try that again tomorrow)

Screen Shot 2022-12-12 at 1 59 25 PM

**Comment by cristipaval** [comment]
@Beamanator Thanks for the review.
Just FYI: This is how it is currently in production. I am happy to change in this PR. 
@shawnborton, what do you think?

**Comment by Beamanator** [comment]
Huh, you're right @cristipaval ! I totally thought the "Phone number" header didn't used to exist if you don't have a phone number secondary login 😅 

But I do think there's a bit more spacing between the "Phone number" header & the button in production - when you mouse over the button you can see it's super close to the "Phone number" header... Think we could at least adjust that?

**Comment by Beamanator** [comment]
Honestly I like that idea a lot @shawnborton 👍

**Comment by cristipaval** [comment]
What do you guys think, @shawnborton @Beamanator ?

https://user-images.githubusercontent.com/18078393/207451769-a1b6bfb5-1722-4dd1-8320-5d4fc3059631.mov
Screenshot 2022-12-13 at 23 47 52

**Comment by shawnborton** [comment]
Ah sorry for the confusion, you had the sizing right, it was just the color we needed to adjust. So something like this:
image

**Comment by cristipaval** [comment]
Sorry @shawnborton ! I re-read you comment and it shouldn't have been confusing but I misunderstood it somehow.
What about now?
Screenshot 2022-12-14 at 23 43 04

**Comment by shawnborton** [comment]
Looks great! 

Now I am noticing that the email address is not in the right color though - it should be in the lighter/brighter text color:
image

**Comment by cristipaval** [comment]
Oh yes @shawnborton , my bad, sorry!

Screenshot 2022-12-15 at 00 10 04

**Comment by sobitneupane** [comment]
@cristipaval Getting following error in mWeb and small screen window which looks unrelated to this PR but it is not reproducible on main. Can you please merge main.
Screenshot 2022-12-15 at 11 08 49

**Comment by cristipaval** [comment]
Thanks @shawnborton !

Now that the look is approved, could you review, @Beamanator @youssef-lr ?

**Comment by cristipaval** [comment]
@sobitneupane yes this looks unrelated to my PR, let me merge main and check.

File: src/pages/settings/Profile/LoginField.js
Comment (Beamanator) [inline src/pages/settings/Profile/LoginField.js:68]: NAB - I feel like we can clean up this whole if-case like this:

```suggestion
        // Check if phone number or email is not validated
        if (this.props.login.partnerUserID && !this.props.login.validatedDate) {
            if (this.props.type === CONST.LOGIN_TYPE.PHONE) {
                note = this.props.translate('loginField.numberHasNotBeenValidated');
            } else {
                note = this.props.translate('loginField.emailHasNotBeenValidated');
            }
        }
        
        
        
        
        
```

File: src/pages/settings/Profile/LoginField.js
Comment (Beamanator) [inline src/pages/settings/Profile/LoginField.js:82]: Sorry to be a bit picky, but can't we still use `MenuItemWithTopDescription` even if `this.props.login.partnerUserID` is undefined? Won't that just mean the title is empty, which is fine? I'd prefer to not have a huge ternary here iffffff possible

File: src/pages/settings/Profile/LoginField.js
Comment (Beamanator) [inline src/pages/settings/Profile/LoginField.js:87]: Seems you're gunna have to clean this up due to the Linter :D But either way I think it's much more readable if we move this logic before the JSX & maybe calculate the title in multiple steps

**Review by Beamanator — CHANGES_REQUESTED** [review]
I know we're going to get rid of `LoginField` eventually so I hate to add more comments on how to clean up that component... But I get a bit confused because it looks like we're checking the existence of `this.props.login.partnerUserID` often, but displaying `this.props.label` - do you think we can use one or the other instead of both? Does my question make sense ? 😅

**Comment by cristipaval** [comment]
> @cristipaval Getting following error in mWeb and small screen window which looks unrelated to this PR but it is not reproducible on main. Can you please merge main.

Hey @sobitneupane ! I merged main and I can't reproduce your issue. Tested on dev and staging.

File: src/pages/settings/Profile/LoginField.js
Comment (cristipaval) [reply src/pages/settings/Profile/LoginField.js:68]: yeah, I agree that even if I didn't touch this code, I should have made it cleaner. I thought it doesn't make sense to refactor the entire class because we're about to get rid off it.
It is now fixed.

File: src/pages/settings/Profile/LoginField.js
Comment (cristipaval) [reply src/pages/settings/Profile/LoginField.js:82]: Currently it is not possible because we have the **Resend** button inside this item. I guess it is not worth modifying `MenuItem` component to make it support Resend button given the fact that we're going to et rid of this one.

File: src/pages/settings/Profile/LoginField.js
Comment (cristipaval) [reply src/pages/settings/Profile/LoginField.js]: fixed.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/25876548/208085276-c476adcc-cc1b-48b9-b87a-5dbe0bb5cece.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/25876548/208085740-b3d6fa64-b466-45e5-809d-2c436e3c5699.mov


</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/25876548/208085315-0201f796-0ace-4b4b-bccf-6490211afcaf.mov

 
</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/25876548/208085371-96135751-6bc5-4f1f-ba23-d77485180139.mov


</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/25876548/208085504-3da07806-5a73-4bd8-b8e6-32cd445263fc.mov

</details>

**Comment by sobitneupane** [comment]
I am getting following console errors:

When clicking on "Pronouns" to open list of pronouns:

Screenshot 2022-12-16 at 16 57 20
Screenshot 2022-12-16 at 16 57 27
Screenshot 2022-12-16 at 16 57 35

On clicking every tabs(Display Name, Pronouns, Timezone):
(I have seen the below console error in other places as well so, not sure if it is related to this PR)
Screenshot 2022-12-16 at 16 57 03

**Comment by cristipaval** [comment]
Hey @sobitneupane ! Thanks for the new review. I fixed the error which was showing when landing on Pronouns page. The other error (which I can confirm that I saw in main as well in the past, so might not be related to this PR) is not reproducible to me after merging main into my branch. Could you test again?

**Review by sobitneupane — COMMENTED** [review]
@cristipaval Similar console errors appear on Timezone list as well. Sorry for not letting you know earlier along with Pronouns list.

Steps:
Profile > TimeZone > TimeZone

**Comment by cristipaval** [comment]
Thanks @sobitneupane ! Could you test again?

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by youssef-lr — APPROVED** [review]
LGTM and tested well! All yours @Beamanator.

**Review by Beamanator — APPROVED** [review]
Looking great! Merge timeee

**Comment by cristipaval** [comment]
This PR introduced [this](https://expensify.slack.com/archives/C049HHMV9SM/p1671686630093889) regression.

**Pull request merged**
